### PR TITLE
feat: let user control whether paste image is confirmed

### PR DESCRIPTION
**Pull request by aaronmcadam**
This makes the process a bit faster and aligns better with Obsidian behaviour.

I've raised this PR, not expecting it to be merged, but to discuss whether it'd be helpful. 

What do you think @epwalsh?

### CODE REVIEW

File: lua/obsidian/img_paste.lua [hunk 1]
```diff
@@ -87,9 +87,10 @@ end
 ---@param fname string|?
 ---@param default_dir obsidian.Path|string
 ---@param default_name string|?
+---@param should_confirm boolean|?
 ---
 ---@return obsidian.Path|? image_path the absolute path to the image file
-M.paste_img = function(fname, default_dir, default_name)
+M.paste_img = function(fname, default_dir, default_name, should_confirm)
```
Comment (epwalsh) [inline lua/obsidian/img_paste.lua:93]:
```suggestion
M.paste_img = function(opts)
```

File: lua/obsidian/img_paste.lua [hunk 2]
```diff
@@ -87,9 +87,10 @@ end
 ---@param fname string|?
 ---@param default_dir obsidian.Path|string
 ---@param default_name string|?
+---@param should_confirm boolean|?
```
Comment (epwalsh) [inline lua/obsidian/img_paste.lua:90]: All 4 `@param` annotations would be condensed to this:
```suggestion
---@param opts { fname: string|?, default_dir: obsidian.Path|string|?, default_name: string|?, should_confirm: boolean|? }|?
```

File: lua/obsidian/img_paste.lua (hunk 1 above)
Comment (epwalsh) [reply lua/obsidian/img_paste.lua:93]: To ensure the table is never `nil`, you could add `opts = opts or {}` at the start of the function.

**Review by epwalsh — COMMENTED** [review]
Hey @aaronmcadam, this is definitely a good addition! I just have a couple minor comments:
- Could you change the config field name to `confirm_img_paste`?
- The `paste_img` function has more positional arguments than I would like now. It would be good to condense these all into a table of options. See my suggestion below.

File: lua/obsidian/img_paste.lua (hunk 2 above)
Comment (aaronmcadam) [reply lua/obsidian/img_paste.lua]: Thanks! I've made these changes!

File: lua/obsidian/img_paste.lua (hunk 1 above)
Comment (aaronmcadam) [reply lua/obsidian/img_paste.lua]: Thanks!

**Comment by aaronmcadam** [comment]
@epwalsh Thanks for the review!

What do you think of letting users avoid this Input dialog too?

CleanShot 2024-02-28 at 11 32 08@2x

This would make pasting images even faster.

**Review by epwalsh — APPROVED** [review]
LGTM, thanks @aaronmcadam!

**Pull request merged**